### PR TITLE
Update box scans for v2.0a + use `TokenId`s from oracle config

### PR DESCRIPTION
**Pull request by kettlebell**

### CODE REVIEW

File: core/src/oracle_state.rs [hunk 1]
```diff
@@ -136,13 +135,15 @@ impl OraclePool {
...
-        let refresh_nft = PoolContract::new().refresh_nft_token_id();
+        let oracle_pool_nft = config.oracle_pool_nft.clone();
+        let refresh_nft = config.refresh_nft.clone();
         let oracle_pool_participant_token_id = config.oracle_pool_participant_token_id.clone();
         let data_point_source = config.data_point_source()?;
 
         let refresh_box_scan_name = "Refresh Box Scan";
-        let datapoint_contract_address = OracleContract::new().ergo_tree();
+        let datapoint_contract_address = OracleContract::new()
+            .with_pool_nft_token_id(oracle_pool_nft.clone())
```
Comment (kettlebell) [inline core/src/oracle_state.rs:145]: Just wanted to check if this is correct. If we change a constant in the script, does it change the address?

Comment (greenhat) [reply core/src/oracle_state.rs:145]: Yes. It changes the ErgoTree by swapping the value in constants.

File: core/src/scans.rs [hunk 2]
```diff
@@ -123,6 +123,10 @@ pub fn register_pool_box_scan(oracle_pool_nft: &TokenId) -> Result<Scan> {
             "predicate": "containsAsset",
             "assetId": oracle_pool_nft.clone(),
         },
+        {
+            "predicate": "containsAsset",
```
Comment (greenhat) [inline core/src/scans.rs:127]: With pool token being NFT (only one exists) I don't see how there could be more than one box with the same pool NFT anyway. Are you sure we need this reward token id filter?

**Review by greenhat — CHANGES_REQUESTED** [review]
Looking good! Please see my comment.

File: core/src/scans.rs (hunk 2 above)
Comment (kettlebell) [reply core/src/scans.rs:127]: Yes of course, thanks for that, we don't need it.

File: core/src/oracle_state.rs (hunk 1 above)
Comment (kettlebell) [reply core/src/oracle_state.rs:145]: Got it, thanks. I think there's a couple of other places where we need to set some constants.

**Review by greenhat — APPROVED** [review]
Looking great! This made me re-thought https://github.com/ergoplatform/oracle-core/issues/30

**Pull request merged**